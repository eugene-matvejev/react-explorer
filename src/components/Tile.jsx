import React from 'react';
import PropTypes from 'prop-types';

const Tile = ({ className, ...props }) =>
    <div
        {...props}
        className={`tile ${className}`}
    />;

Tile.propTypes = {
    className: PropTypes.string,
};

Tile.defaultProps = {
    className: '',
};

export default Tile;