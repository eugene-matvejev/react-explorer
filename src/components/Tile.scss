.rag-tile {
    background: "#4AD964";
    padding: 3em;
    margin: 50px auto 0px;
    width: 300px;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05), 0 0px 40px rgba(0, 0, 0, 0.08);
    border-radius: 5px;

    &:hover {
        box-shadow: 0 3px 8px 0 rgba(0, 0, 0, 0.2);
    }

    &__label {
        text-align: center;
        font-size: 3em;
        font-weight: bold;
    }  
}
