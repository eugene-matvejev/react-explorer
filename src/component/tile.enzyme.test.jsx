import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Tile from './tile';

configure({ adapter: new Adapter() });

describe('<Tile/>', () => {
    const props = {
        to: '//example.com',
    };

    describe('render', () => {
        it('with default/require d props', () => {
            const c = shallow(<Tile {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            test.each`
                prop            | v
                ${'className'}  | ${'{{className}}'}
                ${'data-cy'}    | ${'{{data-cy}}'}
            `(`[::$prop] as "$v"`, ({ prop, v }) => {
                const c = shallow(<Tile {...props} {...{ [prop]: v }} />);

                expect(c).toMatchSnapshot();
            });
        });
    });
});
