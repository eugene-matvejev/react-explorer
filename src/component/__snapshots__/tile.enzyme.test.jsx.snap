// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile/> render with default/require d props 1`] = `
<Link
  className="tile "
  to="//example.com"
/>
`;

exports[`<Tile/> render with optional props [::className] as "{{className}}" 1`] = `
<Link
  className="tile {{className}}"
  to="//example.com"
/>
`;

exports[`<Tile/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<Link
  className="tile "
  data-cy="{{data-cy}}"
  to="//example.com"
/>
`;
